notebooks/plot_benchmark.py

```diff
@@ -33,62 +33,78 @@ def plot_memory_usage(calls_dict, parent_workflow_name, cromwell_id, output_dir)
+                # remove attempt from the task call name
+                name = os.path.dirname(name)
+
             match = re.findall(r"(shard-\d+)$", name)
             # if task is not a shard
             if len(match) == 0:
                 # get the name of the task
                 # i.e: os.path.basename(/workflow/cromwell_id/call) == call
-                task_call = os.path.basename(name)
+                task_name = os.path.basename(name)
 
             # if task is a shard
             else:
-                # get the name of the task that was scattered plus the shard
-                # i.e: os.path.dirname(/workflow/cromwell_id/call/shard) == /workflow_/cromwell_id/call/
-                # i.e: os.path.basename(/workflow/cromwell_id/call) == call
-                task_call = os.path.basename(os.path.dirname(name)) + " " + os.path.basename(name)
-
-            task_calls.append(task_call)
 
-            # get the memory usage and time for each time stamp of the call
-            logs = call["logs"]
-            for log in logs:
-                time = parser.parse(log["time"])
-                time_stamps.append(time)
-
-                memory_usage = log["memory_usage"]
-                memory_usages.append(memory_usage)
-
-            start_time = None
-            if len(time_stamps) > 0:
-                start_time = min(time_stamps)
-
-            start_times.append(start_time)
+                if len(match) > 0:
+                    # get the name of the task that was scattered plus the shard
+                    # i.e: os.path.dirname(/workflow/cromwell_id/call/shard) == /workflow_/cromwell_id/call/
+                    # i.e: os.path.basename(/workflow/cromwell_id/call) == call
+                    task_name = os.path.basename(os.path.dirname(name))
+
+            # if no other shards of the task have been stored in dict
+            if task_name not in tasks.keys():
+                # save name and total memory of task call
+                total_mem = str(call["summary"]["memory"]["size"]) + str(call["summary"]["memory"]["unit"])
+
+                # get max memory usage of task call
+                max_memory = 0.0
+                logs = call["logs"]
+                for log in logs:
+                    memory_usage = log["memory_usage"]
+                    if memory_usage > max_memory:
+                        max_memory = memory_usage
+
+                task_mem_info = {
+                    "max_memory": max_memory,
+                    "total_memory": total_mem}
+                tasks[task_name] = task_mem_info
+
+            # if another other shard of the same task exist
+            else:
+                # update the max memory usage
+                max_memory = tasks[task_name]["max_memory"]
+                for log in logs:
+                    memory_usage = log["memory_usage"]
+                    if memory_usage > max_memory:
+                        max_memory = memory_usage
```

Comment:
There's some duplicate logic between the branches. You could eliminate it by reworking things like:
```python
if task_name not in tasks:
  tasks[task_name] = {"max_memory": 0.0}

# Existing code for when max_memory is already defined...
```